Remove use of asyncio API deprecated in python 3.14

braingram opened the pull request:
~At this moment this also adds python 3.14-dev testing.~ EDIT: This was removed

This remove the use of deprecated asyncio policies.

During local testing with python 3.14 another deprecation was found for `asyncio.iscoroutinefunction`:
https://docs.python.org/3.14/whatsnew/3.14.html#deprecated

This was replaced with `inspect.iscoroutinefunction`.

Fixes: https://github.com/fsspec/filesystem_spec/issues/1861

martindurant commented:
The test environments are based on conda, which doesn't have 3.14dev available yet. We could do a pip-based minimalist CI run without all the optional extras.

braingram commented:
> The test environments are based on conda, which doesn't have 3.14dev available yet. We could do a pip-based minimalist CI run without all the optional extras.

Thanks! I removed that part (and fixed the linting errors). I wanted to wait to see if the other tests passed.

For 3.14 nightly wheels for numpy: https://pypi.anaconda.org/scientific-python-nightly-wheels/ (and possibly other test dependencies) might be needed.

I was able to install an environment locally with python 3.14, dev numpy and just the "test" dependencies. It revealed one more deprecation warning for a single use of `asyncio.iscoroutinefunction` (which is replaced by a function of the same name in inspect that is already in use in other parts of fsspec).

braingram commented:
I think this is ready for review. Let me know if there is any more testing to do, documentation/changelog to update, or anything else.

martindurant commented:
I'll update the changelog at release time :)

Adding a 3.14dev CI run would still be a good idea, but it can be done in a different PR or wait for conda (which will only happen some time after official release).
